poly2tri/common/utils.h

```diff
@@ -68,7 +68,10 @@ Orientation Orient2d(const Point& pa, const Point& pb, const Point& pc)
   double detleft = (pa.x - pc.x) * (pb.y - pc.y);
   double detright = (pa.y - pc.y) * (pb.x - pc.x);
   double val = detleft - detright;
-  if (val > -EPSILON && val < EPSILON) {
+
+// Using a tolerance here fails on concave-by-subepsilon boundaries
+//   if (val > -EPSILON && val < EPSILON) {
+  if (val == 0) {
```

Comment:
Maybe use `std::fpclassify(val) == FP_ZERO` to avoid the float comparision here?

Reply:
Exactly :)

I think some compilers will warn on `equal_to`, too. But I'm not sure.